MyStore/src/GeneratorData.cpp

```diff
@@ -0,0 +1,35 @@
+#include "GeneratorData.h"
+
+const int period_for_element_milliseconds = 1;
+
+
+std::vector<InfoProduct> GeneratorData::DoGenarate()
+{
+	InfoProduct data;
+
+    for (int i = 0; i < m_colors.size(); i++)
+	{
+		if (m_data.size() == m_max_amount_data)
+			break;
+	    data.m_product_color = m_colors[i];
+
+		for (int j = 0; j < m_names.size(); j++)
+	{
+			if (m_data.size() == m_max_amount_data)
+				break;
+			data.m_product_name = m_names[j];
+		    m_data.push_back(data);
+			std::this_thread::sleep_for(std::chrono::milliseconds(period_for_element_milliseconds));

```

Comment:
As we discussed, rework this part to get rid of sleep and to generate only one product at a time. Also pick random color and name from arrays when generating product.